HACKING.md

```diff
@@ -244,23 +244,14 @@ you are never committing anything with a whitespace violation:
 Type-checking the README directory
 ----------------------------------
 
-* By default the README directory is not exported in the
-  `standard-library.agda-lib` file in order to avoid clashing with other people's
-  README files. This means that by default type-checking a file in the README
-  directory fails.
-
-* If you wish to type-check a README file, then you will need to change the line:
-  ```
-  include: src
-  ```
-  to
-  ```
-  include: src .
-  ```
-  in the `standard-library.agda-lib` file.
-
-* Please do not include this change in your pull request.
-
+* By default the README files are not exported in the
+  `standard-library.agda-lib` file in order to avoid
+  clashing with other people's README files.
+
+* If you wish to type-check a README file, then you will
+  need to change the present working directory to `doc/`
```

Comment:
So, I've been thinking a lot (if 'hard') about this, and I don't know if it makes more, or less, sense, to have the `README.*` modules in a `doc` directory, or in good-old `README` as a top-level directory, and simply bump all the modules names 'up' one level, by having what would then be `README.agda` import from `Axiom` rather than `README.Axiom` etc.

Alternatively, leave the *entire* existing `README/` directory alone, but simply have *its* `blah.agda-lib` point at... the right directories (and here is where I get confused about what those should be... `. ../src` as already, or `. ./README ../src` (I guess this latter is subsumed by the former?)

In other words, the change could be less invasive still simply by moving toplevel `README.agda` to `README/`, and updating the `GNUmakefile` and `.github/tooling` scripts?

Reply:
Many modules in `README` have identical paths as the modules from the standard library (e.g. `README/Function/Reasoning.agda` and `src/Function/Reasoning.agda`, `README/Data/List.agda` and `src/Data/List.agda`, etc.). If the `README.` prefix in the module paths is removed, I wonder how Agda resolves the module?

Reply:
I think because [the `include` directive line](https://github.com/agda/agda-stdlib/blob/561d92420ce6f318b17d431530e1ccdaa6a04263/doc/standard-library-doc.agda-lib#L2)
ensures that those in `doc/README` will get searched for before trying to look for them in `src`?

Reply:
The `include` in this PR looks good to me. What I don't understand is how your alternative approach works. (this one:
> ... in good-old README as a top-level directory, and simply bump all the modules names 'up' one level, by having what would then be README.agda import from Axiom rather than README.Axiom etc.
> 
> Alternatively, leave the entire existing README/ directory alone, but simply have its blah.agda-lib point at... the right directories (and here is where I get confused about what those should be... . ../src as already, or . ./README ../src (I guess this latter is subsumed by the former?)

I might be wrong, but including `../README` directory before `../src` in `include:` sounds like modules in `README` will shadow those that have the same name in `src`. Without the `README.` prefix in the module paths, how could files like `README.Data.List.Relation.Binary.Equality` that import `Data.Nat` continue to work? Wouldn't `Data.Nat` incorrectly refers to `README.Data.Nat`?

Reply:
Ah... ok, perhaps you're right. 

In any case, I would prefer simply to have the `doc` directory be called `README` instead. Or are you saying that such a solution also would not work?

Reply:
I think that should work.